client/src/views/files/FileOperationMenu.vue

```diff
@@ -197,8 +203,19 @@ function getFileOperationComponent(item: OperationItem): Component | undefined {
 }
 
 function updateImportState(id: string, state: FileOperationState, stateData?: StateData): void {
-  const index = inProgressItems.value.findIndex((op) => op.data.id === id);
+  let index: number;
+
+  if (state === FileOperationState.OperationProgress) {
+    // Check if item has to be moved from queued
+    index = queuedItems.value.findIndex((op) => op.data.id === id);
```

Comment:
I would do the opposite: check if the item is already in `inProgressItems`, if not push it and then try to remove it from queuedItems.

The reason being that `inProgressItems` will probably be quite short (it shouldn't contain more than MAX_SIMULTANEOUS_OPERATIONS) which means a find() would be very quick, whereas `queuedItems` could become very big.